misc/init.d/cfengine3.in

```diff
@@ -348,10 +348,18 @@ cf_stop_daemon()
         fi
 
         pidfile_pid="$(cf_pidfile_pid "$pidfile")"
+        matching_pids="$(cf_get_matching_pids "$base_daemon")"
         if [ -n "$pidfile_pid" ]; then
-            pids="$pidfile_pid"
+            if ! echo "$matching_pids" | grep "$pidfile_pid" >/dev/null; then
```

Comment:
It looks like `cf_get_matching_pids` [prints pids one-per-line][1], so to avoid grep matching "1234" when looking for "123" let's add `^` and `$` to the grep command, like this: `grep "^$pidfile_pid\$"`

[1]: https://github.com/cfengine/core/blob/a8d14a7721c53008725bfdd25f440b7a316b10bd/misc/init.d/cfengine3.in#L227

Reply:
Good catch!